// base-shell/es/containers/Layout/Layout.js
import _regeneratorRuntime from "@babel/runtime/regenerator";

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

import LocaleProvider from '../../providers/Locale/Provider';
import React, { Suspense, useEffect, useState } from 'react';
import { IntlProvider } from 'react-intl';
import { getLocaleMessages } from '../../utils/locale';
import { useLocale } from '../../providers/Locale';
import { useRoutes } from 'react-router-dom';
import UpdateProvider from '../../providers/Update/Provider';
import AuthProvider from '../../providers/Auth/Provider';
import AddToHomeScreenProvider from '../../providers/AddToHomeScreen/Provider';
import OnlineProvider from '../../providers/Online/Provider';
import SimpleValuesProvider from '../../providers/SimpleValues/Provider';
import { useConfig } from '../../providers/Config';
export var LayoutContent = function LayoutContent(_ref) {
  var _ref$appConfig = _ref.appConfig,
      appConfig = _ref$appConfig === void 0 ? {} : _ref$appConfig;

  var _useState = useState([]),
      messages = _useState[0],
      setMessages = _useState[1];

  var _ref2 = appConfig || {},
      components = _ref2.components,
      _ref2$routes = _ref2.routes,
      routes = _ref2$routes === void 0 ? [] : _ref2$routes,
      containers = _ref2.containers,
      confLocale = _ref2.locale,
      getDefaultRoutes = _ref2.getDefaultRoutes,
      auth = _ref2.auth,
      update = _ref2.update;

  var _ref3 = auth || {},
      persistKey = _ref3.persistKey;

  var _ref4 = update || {},
      _ref4$checkInterval = _ref4.checkInterval,
      checkInterval = _ref4$checkInterval === void 0 ? 5000 : _ref4$checkInterval;

  var _ref5 = components || {},
      Menu = _ref5.Menu,
      _ref5$Loading = _ref5.Loading,
      Loading = _ref5$Loading === void 0 ? function () {
    return /*#__PURE__*/React.createElement("div", null, "Loading...");
  } : _ref5$Loading;

  var _ref6 = confLocale || {},
      locales = _ref6.locales,
      onError = _ref6.onError;

  var _ref7 = containers || {},
      _ref7$LayoutContainer = _ref7.LayoutContainer,
      LayoutContainer = _ref7$LayoutContainer === void 0 ? React.Fragment : _ref7$LayoutContainer;

  var defaultRoutes = getDefaultRoutes ? getDefaultRoutes(appConfig) : [];

  var _useLocale = useLocale(),
      _useLocale$locale = _useLocale.locale,
      locale = _useLocale$locale === void 0 ? {} : _useLocale$locale;

  useEffect(function () {
    var loadPolyfills = /*#__PURE__*/function () {
      var _ref8 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
        var i, l;
        return _regeneratorRuntime.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                if (!(locale.locales && locale.locales.length > 0)) {
                  _context.next = 11;
                  break;
                }

                i = 0;

              case 2:
                if (!(i < locales.length)) {
                  _context.next = 11;
                  break;
                }

                l = locales[i];

                if (!(l.locale === locale)) {
                  _context.next = 8;
                  break;
                }

                if (!l.loadData) {
                  _context.next = 8;
                  break;
                }

                _context.next = 8;
                return l.loadData;

              case 8:
                i++;
                _context.next = 2;
                break;

              case 11:
              case "end":
                return _context.stop();
            }
          }
        }, _callee);
      }));

      return function loadPolyfills() {
        return _ref8.apply(this, arguments);
      };
    }();

    loadPolyfills();
  }, [locale, locales]);
  useEffect(function () {
    var loadMessages = /*#__PURE__*/function () {
      var _ref9 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee2() {
        var messages;
        return _regeneratorRuntime.wrap(function _callee2$(_context2) {
          while (1) {
            switch (_context2.prev = _context2.next) {
              case 0:
                _context2.next = 2;
                return getLocaleMessages(locale, locales);

              case 2:
                messages = _context2.sent;
                setMessages(messages);

              case 4:
              case "end":
                return _context2.stop();
            }
          }
        }, _callee2);
      }));

      return function loadMessages() {
        return _ref9.apply(this, arguments);
      };
    }();

    loadMessages();
  }, [locale, locales]);
  return /*#__PURE__*/React.createElement(AuthProvider, {
    persistKey: persistKey
  }, /*#__PURE__*/React.createElement(SimpleValuesProvider, null, /*#__PURE__*/React.createElement(AddToHomeScreenProvider, null, /*#__PURE__*/React.createElement(UpdateProvider, {
    checkInterval: checkInterval
  }, /*#__PURE__*/React.createElement(OnlineProvider, null, /*#__PURE__*/React.createElement(IntlProvider, {
    locale: locale,
    key: locale,
    messages: messages,
    onError: onError
  }, /*#__PURE__*/React.createElement(LayoutContainer, null, /*#__PURE__*/React.createElement(Suspense, {
    fallback: /*#__PURE__*/React.createElement(Loading, null)
  }, Menu && /*#__PURE__*/React.createElement(Menu, null)), /*#__PURE__*/React.createElement(Suspense, {
    fallback: /*#__PURE__*/React.createElement(Loading, null)
  }, useRoutes([].concat(routes, defaultRoutes))))))))));
};
export var Layout = function Layout() {
  var _useConfig = useConfig(),
      appConfig = _useConfig.appConfig;

  var _ref10 = appConfig || {},
      locale = _ref10.locale;

  var _ref11 = locale || {},
      defaultLocale = _ref11.defaultLocale,
      persistKey = _ref11.persistKey;

  return /*#__PURE__*/React.createElement(LocaleProvider, {
    defaultLocale: defaultLocale,
    persistKey: persistKey
  }, /*#__PURE__*/React.createElement(LayoutContent, {
    appConfig: appConfig
  }));
};
export default Layout;
