// base-shell/es/components/AuthorizedRoute.js
import React from "react";
import { Navigate, useLocation } from "react-router-dom";
import { useAuth } from "../providers/Auth";
import { useConfig } from "../providers/Config";

function AuthorizedRoute(_ref) {
  var children = _ref.children;

  var _useConfig = useConfig(),
      appConfig = _useConfig.appConfig;

  var _ref2 = appConfig || {},
      authConfig = _ref2.auth;

  var _ref3 = authConfig || {},
      _ref3$signInURL = _ref3.signInURL,
      signInURL = _ref3$signInURL === void 0 ? "/signin" : _ref3$signInURL;

  var _useAuth = useAuth(),
      auth = _useAuth.auth;

  var location = useLocation();

  if (auth.isAuthenticated) {
    return children;
  } else {
    return /*#__PURE__*/React.createElement(Navigate //to={`${signInURL}?from=${location.pathname}`}
    , {
      to: {
        pathname: signInURL,
        search: "from=" + location.pathname,
        state: {
          from: location
        }
      },
      replace: true
    });
  }
}

export default AuthorizedRoute;
