// base-shell/es/containers/App/App.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { Suspense, lazy } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import ConfigProvider from '../../providers/Config/Provider';
var Layout = lazy(function () {
  return import('../../containers/Layout/Layout');
});

var App = function App(_ref) {
  var appConfig = _ref.config;

  var config = _extends({}, appConfig);

  var pages = config.pages,
      components = config.components,
      containers = config.containers;

  var _ref2 = pages || {},
      _ref2$LandingPage = _ref2.LandingPage,
      LandingPage = _ref2$LandingPage === void 0 ? false : _ref2$LandingPage;

  var _ref3 = components || {},
      _ref3$Loading = _ref3.Loading,
      Loading = _ref3$Loading === void 0 ? function () {
    return /*#__PURE__*/React.createElement("div", null);
  } : _ref3$Loading;

  var _ref4 = containers || {},
      _ref4$AppContainer = _ref4.AppContainer,
      AppContainer = _ref4$AppContainer === void 0 ? React.Fragment : _ref4$AppContainer;

  return /*#__PURE__*/React.createElement(Suspense, {
    fallback: /*#__PURE__*/React.createElement(Loading, null)
  }, /*#__PURE__*/React.createElement(ConfigProvider, {
    appConfig: config
  }, /*#__PURE__*/React.createElement(AppContainer, null, /*#__PURE__*/React.createElement(BrowserRouter, null, /*#__PURE__*/React.createElement(Routes, null, LandingPage && /*#__PURE__*/React.createElement(Route, {
    path: "/",
    exact: true,
    element: /*#__PURE__*/React.createElement(LandingPage, null)
  }), /*#__PURE__*/React.createElement(Route, {
    path: "*",
    element: /*#__PURE__*/React.createElement(Suspense, {
      fallback: /*#__PURE__*/React.createElement(Loading, null)
    }, /*#__PURE__*/React.createElement(Layout, {
      appConfig: config
    }))
  }))))));
};

export default App;
